build.gradle

```diff
@@ -102,14 +103,61 @@ task configureGroovyAll {
             }
         }
     }
+
+    doLast {
+        def expandedArchives = file("${buildDir}/tmp/expandedArchives")
+        def mergedFolder = new File(expandedArchives, 'merged')
+
+        collectDuplicates(expandedArchives.listFiles())
+            .each { String relative, List<File> files ->
+                def mergedFile = new File(mergedFolder, relative)
+                mergedFile.parentFile.mkdirs()
+                mergedFile.text = merge(relative, files)
+                files*.delete() // TODO doesn't work (manifests are read after this, and even if we delete, they'll still end up in the jar)
+            }
+    }
+}
+
+static collectDuplicates(File[] roots) {
+    Map<String, List<File>> duplicates = [:].withDefault { [] }
+    roots.each { root ->
+        root.eachFileRecurse(FileType.FILES) {
+            def relative = root.toURI().relativize(it.toURI())
+            duplicates[relative.toString()] += it
+        }
+    }
+    return duplicates.findAll { it.value.size() > 1 }
+}
+
+static merge(String relative, List<File> files) {
+    switch (relative) {
+        case ['META-INF/LICENSE', 'META-INF/NOTICE', 'META-INF/services/org.codehaus.groovy.transform.ASTTransformation']:
```

Comment:
For me it would be easier to parse this notation:

```groovy
case 'META-INF/LICENSE':
case 'META-INF/NOTICE':
case 'META-INF/services/org.codehaus.groovy.transform.ASTTransformation':
```

Reply:
Changed it, let me know if the Groovy syntax is still too much here

Reply:
Done